tests/app/main/views/test_buyers.py

```diff
@@ -0,0 +1,50 @@
+import mock
+from ...helpers import LoggedInApplicationTest
+from lxml import html
+
+
+@mock.patch('app.main.views.buyers.data_api_client')
+class TestBuyersView(LoggedInApplicationTest):
+
+    def test_should_be_a_404_if_no_brief_found(self, data_api_client):
+        data_api_client.get_brief.return_value = None
+        response = self.client.get('admin/buyers?brief_id=1')
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_should_display_a_useful_message_if_no_brief_found(self, data_api_client):
+        data_api_client.get_brief.return_value = None
+        response = self.client.get('admin/buyers?brief_id=1')
+
+        document = html.fromstring(response.get_data(as_text=True))
+        banner_message = document.xpath('//p[@class="banner-message"]//text()')[0].strip()
+
+        self.assertEqual("Sorry, we couldn't find a brief with the supplied ID: 1", banner_message)
+
+    def test_table_should_show_a_useful_message_if_no_users(self, data_api_client):
```

Comment:
I'm not sure this should ever be able to happen (i.e. a brief with no users)?
And in that case if it ever did happen then maybe the message should be more dramatic.  Or have I missed a possible case where a brief with no users would be legit?